core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -891,7 +944,7 @@ class ReplicaManager(val config: KafkaConfig,
     def processFailedRecord(topicPartition: TopicPartition, t: Throwable) = {
       val logStartOffset = getPartition(topicPartition) match {
         case HostedPartition.Online(partition) => partition.logStartOffset
-        case HostedPartition.None | HostedPartition.Offline => -1L
+        case _ => -1L
```

Comment:
I think you want to call `onlinePartition` instead of `getPartition`.

Reply:
Agreed. Maybe `onlinePartition(topicPartition).map(_.logStartOffset).getOrElse(-1L)`?